File: app/src/main/java/org/stepic/droid/core/ShareHelper.java
package org.stepic.droid.core;

import android.content.Intent;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import org.stepic.droid.model.CertificateViewItem;
import org.stepic.droid.model.Course;
import org.stepic.droid.model.Lesson;
import org.stepic.droid.model.Step;
import org.stepic.droid.model.Unit;

import javax.inject.Singleton;

@Singleton
public interface ShareHelper {
    Intent getIntentForCourseSharing(@NotNull Course mCourse);

    Intent getIntentForShareCertificate(@NotNull CertificateViewItem certificateViewItem);

    Intent getIntentForStepSharing(Step step, Lesson lesson, @Nullable Unit unit);
}

File: app/src/main/java/org/stepic/droid/core/ShareHelper.kt
package org.stepic.droid.core

import android.content.Intent
import org.stepic.droid.model.CertificateListItem
import org.stepik.android.model.*
import org.stepik.android.model.Unit
import org.stepik.android.model.user.User

interface ShareHelper {
    fun getIntentForCourseSharing(course: Course): Intent

    fun getIntentForShareCertificate(certificateListItem: CertificateListItem.Data): Intent

    fun getIntentForStepSharing(step: Step, lesson: Lesson, unit: Unit?): Intent

    fun getIntentForSectionSharing(section: Section): Intent

    fun getIntentForUserSharing(user: User): Intent

    fun getIntentForCourseResultSharing(course: Course, message: String): Intent

    fun getIntentForCourseResultCertificateSharing(certificate: Certificate, message: String): Intent
}
